### DISCUSSION TITLE
High GPU usage, and higher cpu usage on infinite animation even when the object is hidden

### DISCUSSION THREAD

**Discussion by Shivansps** [AvaloniaUI/Avalonia#15389] [Q&A]
I wanted to have a small rotating icon to indicate currently running tasks, but the resource usage of this seems to be insane
Here is the code in question the animation currently is hidden as object visibility is set to false
![high_cpu1](https://github.com/AvaloniaUI/Avalonia/assets/2444336/7a608f5c-3ab5-41a5-99ca-5f4e1c67cf68)
as you can see 6% gpu usage and 1% cpu usage on nothing i can see, and thats still a lot for just one small single rotating icon of a 32x32 192 byte png,

im doing something wrong? at the point it might be better to have like 10 images and manually do the animation by switching their visibility

He is the baseline, the app on idle does nothing normally, 0% cpu and gpu.
![low_cpu](https://github.com/AvaloniaUI/Avalonia/assets/2444336/4d35d178-c27a-4857-9ba4-edb63add8c10)

**Comment by timunie** [comment]
see also #12922

**Reply by Shivansps** [reply]
mmm not sure if this is a memory leak, i dont see the memory increasing over time.
The problem is that the animation uses far too much resources, i see up to 15% (of a RTX3060) for a small rotating icon, 1% cpu is also a lot for that, and all this even while the object visibility is set to false too.

So i wanted to ask for alternative ways to do it.
